### ISSUE TITLE
Add a recommended add-on to the list

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-server#11064]
Currently we have the recommended add-on list used by TAAR and return to AMO this list is a good place to start.

We can continue to use this list, but we will need to be able to do the following:

* A newly recommended add-on will need to be placed into a "pending recommendation" state.
* The next version uploaded will be pushed into a different review queue (the review flow is a independent project), the next version will be signed as recommended only once the review has passed.
* Unlisted add-ons can be recommended

For more details see the PRD: https://docs.google.com/document/d/1fRxLy7h02tn1HBIlsTU_6h9DHmN8c3N6bg4JoxHF7J8/edit#heading=h.pbx0vxiwi6qq

During the changes we must ensure that the TAAR and RTAMO lists continue to function.

**Comment by bobsilverberg** [comment]
Subscribing

**Comment by eviljeff** [comment]
Moving "~Unlisted add-ons~ Add-ons with only unlisted versions can be recommended" to a separate issue because we've got to be careful that non-public add-ons aren't included on the discovery pane regardless of them being recommended.

**Comment by eviljeff** [comment]
QA: it's going to be difficult to test this issue in isolation - you can alter the status in admin but the outcome of new versions being available for review are only going to be testable once the reviewer tools issues are completed.

**Comment by AlexandraMoga** [comment]
@eviljeff I've noticed that the recommended status can be added by marking the checkbox in the Discovery admin tool - https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/discoveryitem/3/change/ . 
This change will be then reflected in the curated list by the 'Recommendable' column.

As you mentioned above, there is not much left to test at this point, since the effects of this change depend on other mechanisms to be implemented.

I'm not adding a verified label to this issue, since not all the requirements are available for testing.
